### PR TITLE
Random walker bug fix: no error should be raised when there is nothing to do

### PR DESCRIPTION
**Pull request by emmanuelle**
This PR fixes the bug reported by #3473, for the specific case when there is nothing to do (only already labeled pixels, or pixels which cannot be determined because they are not surrounded by masked pixels).

@winston-li

### CODE REVIEW

**Comment by sciunto** [comment]
Just one quick and one side comment.

Quick one. Looks like this PR starts to introduce duplicated codes. Have you tried to figure out if it is possible to refactor that? I have not tried myself... I stopped myself at "alright, the patch looks appropriate" :)

Side comment. The docstring has missing ", optional" starting from beta to the end of the parameters list.

**Comment by emmanuelle** [comment]
@jni and @sciunto I wrote a helper function to avoid duplicated code, and I also added a References section in the docstring. Hope it's ready now!

**Comment by emmanuelle** [comment]
@sciunto don't be shy, you can ask it for yourself :-). I added a minimal docstring.

File: skimage/segmentation/random_walker_segmentation.py [hunk 1]
```diff
@@ -337,6 +351,12 @@ def random_walker(data, labels, beta=130, mode='bf', tol=1.e-3, copy=True,
     This linear system is solved in the algorithm using a direct method for
     small images, and an iterative method for larger images.
 
+    References
+    ----------
+    .. [1] Leo Grady, Random walks for image segmentation, IEEE Trans Pattern
+    Anal Mach Intell. 2006 Nov;28(11):1768-83. doi:10.1109/TPAMI.2006.233
```
[the remark this answers is not in the document]
Comment (sciunto) [reply skimage/segmentation/random_walker_segmentation.py]: don't forget to click here :)

**Comment by sciunto** [comment]
@emmanuelle It looks like there are some pep8 issues. Are they coming from your editor?

File: skimage/segmentation/random_walker_segmentation.py [hunk 2]
```diff
@@ -337,6 +356,12 @@ def random_walker(data, labels, beta=130, mode='bf', tol=1.e-3, copy=True,
     This linear system is solved in the algorithm using a direct method for
     small images, and an iterative method for larger images.
 
+    References
+    ----------
+    .. [1] Leo Grady, Random walks for image segmentation, IEEE Trans Pattern
+    Anal Mach Intell. 2006 Nov;28(11):1768-83. DOI:`10.1109/TPAMI.2006.233`
```
Comment (sciunto) [inline skimage/segmentation/random_walker_segmentation.py]:
```suggestion
    Anal Mach Intell. 2006 Nov;28(11):1768-83. :DOI:`10.1109/TPAMI.2006.233`
```
my bad... sorry.

File: skimage/segmentation/random_walker_segmentation.py [hunk 3]
```diff
@@ -337,6 +356,11 @@ def random_walker(data, labels, beta=130, mode='bf', tol=1.e-3, copy=True,
     This linear system is solved in the algorithm using a direct method for
     small images, and an iterative method for larger images.
 
+    References
+    ----------
+    .. [1] Leo Grady, Random walks for image segmentation, IEEE Trans Pattern
+    Anal Mach Intell. 2006 Nov;28(11):1768-83. :DOI:`10.1109/TPAMI.2006.233`.
```
Comment (sciunto) [inline skimage/segmentation/random_walker_segmentation.py]: I think this line must be indented, causing the travis failure.

Comment (emmanuelle) [reply skimage/segmentation/random_walker_segmentation.py]: thanks I was getting crazy over this failure!

Comment (sciunto) [reply skimage/segmentation/random_walker_segmentation.py]: you're welcome :)

I'm not sure it is indented enough. I don't think sphinx consider indents as 4 spaces like in python, but rather an alignment with the L of Leo. At least, this is how we do elsewhere in the library.

Sphinx is particularly annoying on that, and hard to debug.

Comment (emmanuelle) [reply skimage/segmentation/random_walker_segmentation.py]: Yes I hesitated between the two ways of indenting but Travis is all green now, so I'd prefer not over-indenting.

**Comment by emmanuelle** [comment]
@jni your requests were addressed!

**Comment by jni** [comment]
@emmanuelle indeed they were! =)

**Pull request merged**